src/protocol/securemul.rs

```diff
@@ -14,6 +16,53 @@ pub struct DValue<F> {
     d: F,
 }
 
+#[derive(Debug)]
+pub struct SecureMulAll<'a, N> {
+    prss: &'a PrssSpace,
+    gateway: &'a Gateway<N>,
+    step: &'a UniqueStepId,
+}
+
+impl<'a, N: Network> SecureMulAll<'a, N> {
+    pub fn new(prss: &'a PrssSpace, gateway: &'a Gateway<N>, step: &'a UniqueStepId) -> Self {
+        Self {
+            prss,
+            gateway,
+            step,
+        }
+    }
+
+    pub async fn execute<F: Field>(
+        self,
+        a: Batch<Replicated<F>>,
+        b: Batch<Replicated<F>>,
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
```

Comment:
would it be better to return `Batch<F>` ?

Reply:
Or `impl Iterator<Item = Replicated<F>>`